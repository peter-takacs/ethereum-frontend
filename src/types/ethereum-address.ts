export class Address {
    private readonly address: string;

    public static isValid(address: string) {
        return /^0x[a-fA-F0-9]{40}$/.test(address);
    }

    public static fromBytes(bytes: Uint8Array): Address {
        if (bytes.length !== 20) {
            throw new Error(`Invalid amount of bytes supplied to ethereum address.`)
        }
        let address = '0x';
        for (const byte of bytes) {
            address = address + byte.toString(16);
        }
        return new Address(address);
    }

    constructor (address: string) {
        if (!Address.isValid(address)) {
            throw new Error(`Invalid address "${address}" supplied to ethereum address constructor.`)
        }
        this.address = address;
    }

    public toString() {
        return this.address;
    }

    public equals(other: Address | null) {
        if (other == null) {
            return false;
        }
        return this.address === other.address;
    }
}