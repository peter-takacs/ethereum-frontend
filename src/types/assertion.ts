import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}