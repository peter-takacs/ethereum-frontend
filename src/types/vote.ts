import { Address } from "./ethereum-address";

export enum VoteStatus {
    Pending = 0,
    Accept = 1,
    Reject = 2
}

export interface CandidateVotes {
    candidate: Address;
    votes: Map<Address, VoteStatus>;
}