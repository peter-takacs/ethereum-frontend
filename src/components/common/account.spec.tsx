import * as React from 'react';
import AccountDisplay from './account';
import * as renderer from 'react-test-renderer';
import { Address } from '../../types/ethereum-address';
import { dummyAccount } from '../../utils/test/address';

test('Renders with correct account', () => {
    const component = renderer.create(
        <AccountDisplay account={dummyAccount(0)} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('Renders with empty account', () => {
    const component = renderer.create(
        <AccountDisplay account={null} />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
