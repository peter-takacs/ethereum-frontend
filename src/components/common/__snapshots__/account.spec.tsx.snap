// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with correct account 1`] = `
<div>
  Current account is: 
  0xdf3f619804a92fdb4057192dc43dd748ea778adc
</div>
`;

exports[`Renders with empty account 1`] = `
<div>
  Current ETH account is unavailable.
</div>
`;
